### PR TITLE
Add a Warning Message for Invalid Accident Year Types

### PR DESCRIPTION
**Pull request by nnikolopoulos**
Log <strike>a warning</strike> an error when the input of `AccidentYear` does not match the expected type (**int**). 
This PR closes issue #240.

### CODE REVIEW

File: ifrs17/Constants/Validations.ipynb [hunk 1]
```diff
@@ -194,6 +194,7 @@
                 "\n    (Warning.VariablesAlreadyImported       , 0) => $\"The import of the current file does not contain any new data. Hence, no data will be saved or calculations will be performed.\",",
                 "\n    (Warning.MandatoryAocStepMissing        , 3) => $\"The AoC step ({s[0]}, {s[1]}) is not imported for ({s[2]}).\",",
                 "\n    (Warning.ScenarioReCalculations         , 1) => $\"The present Best Estimate import makes the result of dependent Scenarios out of date. Hence, the following Scenarios are re-calculated: {s[0]}.\", ",
+                "\n    (Warning.AccidentYearTypeNotValid       , 1) => $\"The parsed AccidentYear type {s[0]} is invalid. Expected Accident Year input of type int.\",",
```
Comment (dcolleoni) [inline ifrs17/Constants/Validations.ipynb:197]: Error instead

File: ifrs17/Import/Importers.ipynb [hunk 2]
```diff
@@ -1368,13 +1368,17 @@
                 "\n                if(values.Length == 0 && !parsingStorage.MandatoryAocSteps.Contains(new AocStep(aocType, novelty))) return null;",
                 "\n            }",
                 "\n            ",
+                "\n            var accidentYearInput = (datarow.Field<string>(nameof(RawVariable.AccidentYear)));",
+                "\n            if(hasAccidentYearColumn) ",
```
Comment (dcolleoni) [inline ifrs17/Import/Importers.ipynb:1372]: first thing, check if there is the column

File: ifrs17/Import/Importers.ipynb [hunk 3]
```diff
@@ -1368,13 +1368,17 @@
...
+                "\n                if(!Int32.TryParse(accidentYearInput, out _)) ApplicationMessage.Log(Warning.AccidentYearTypeNotValid, accidentYearInput);",
```
Comment (dcolleoni) [inline ifrs17/Import/Importers.ipynb:1373]: this logs an error when it's null but it should not

File: ifrs17/Import/Importers.ipynb [hunk 4]
```diff
@@ -1368,13 +1368,17 @@
...
+                "\n",
                 "\n            var item = new RawVariable {",
                 "\n                DataNode = dataNode,",
                 "\n                AocType = aocType,",
                 "\n                Novelty = novelty,",
                 "\n                AmountType = valueType.AmountType,",
                 "\n                EstimateType = valueType.EstimateType,",
-                "\n                AccidentYear = hasAccidentYearColumn && Int32.TryParse((datarow.Field<string>(nameof(RawVariable.AccidentYear))), out var accidentYear)",
+                "\n                AccidentYear = hasAccidentYearColumn && Int32.TryParse(accidentYearInput, out var accidentYear)",
```
Comment (dcolleoni) [inline ifrs17/Import/Importers.ipynb:1381]: You should TryParse only once

File: ifrs17/Import/Importers.ipynb (hunk 2 above)
Comment (nnikolopoulos) [reply ifrs17/Import/Importers.ipynb]: I modified the structure a bit, but still the vars are declared before the if statement for the `hasAccidentYearColumn`, does it look okay?

**Pull request closed without merging**